File: intermediate-package/bonus-webinars/writing-parameterized-tests-junit5/src/test/java/com/testwithspring/intermediate/ValueSourceExampleTest.java
package com.testwithspring.intermediate;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import static org.junit.jupiter.api.Assertions.assertNotNull;

@DisplayName("@ValueSource example")
class ValueSourceExampleTest {

    @DisplayName("Should pass the method parameters provided by theValueSource annotation")
    @ParameterizedTest(name = "{index} => message=''{0}''")
    @ValueSource(strings = {"Hello", "World"})
    void shouldPassNotNullMessageAsMethodParameter(String message) {
        assertNotNull(message);
    }
}

File: master-package/bonus-webinars-2019/writing-parameterized-tests-with-kotlin/src/test/kotlin/com/testwithspring/master/kotlin/ValueSourceExampleTest.kt
package com.testwithspring.master.kotlin

import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Tag

@Tag("unitTest")
@DisplayName("@ValueSource example")
class ValueSourceExampleTest {

    @DisplayName("Should pass the function parameters provided by the @ValueSource annotation")
    @ParameterizedTest(name = "{index} => message=''{0}''")
    @ValueSource(strings = ["Hello", "World"])
    fun shouldPassNotNullMessageAsFunctionParameter(message: String) {
        assertNotNull(message)
    }
}